contract/models/contract.py

```diff
@@ -144,11 +153,18 @@ def _onchange_recurring_invoices(self):
         if self.date_start and self.recurring_invoices:
             self.recurring_next_date = self.date_start
 
-    @api.model
+    @api.multi
```

Comment:
Don't change this, as you're passing the line as argument, so there's the possibility that line.analytic_account_id != self. Use always the first one instead for referencing the contract.